edge/services/helloworld/README.md

```diff
@@ -1,77 +1,105 @@
-1. Set your exchange org:
-```
-export HZN_ORG_ID="PUT-YOUR-CLUSTER-NAME-HERE"
-```
-2. Set your exchange user credentials in the Horizon-supported environment variable and verify it:
+- [Preconditions for Using the Hello World Example Edge Service](#preconditions)
+- [Using the Hello World Example Edge Service with Deployment Pattern](#using-helloworld-pattern)
+- [Using the Hello World Example Edge Service with Deployment Policy](#using-helloworld-policy)
+- [Creating Your Own Hello World Edge Service](CreateService.md)
+- Further Learning - to see more Horizon features demonstrated, continue on to the [cpu2msghub example](../../evtstreams/cpu2evtstreams).
+
+## <a id=preconditions></a> Preconditions for Using the Hello World Example Edge Service
+
+If you haven't done so already, you must do these steps before proceeding with the helloworld example:
+
+1. Install the Horizon management infrastructure (exchange and agbot).
+
+2. Install the Horizon agent on your edge device and configure it to point to your Horizon exchange.
+
+3. Set your exchange org:
+
+```bash
+export HZN_ORG_ID="<your-cluster-name>"
 ```
-export HZN_EXCHANGE_USER_AUTH="iamapikey:PUT-YOUR-API-KEY-HERE"
+
+4. Create a cloud API key that is associated with your Horizon instance, set your exchange user credentials, and verify them:
+
+```bash
+export HZN_EXCHANGE_USER_AUTH="iamapikey:<your-API-key>"
 hzn exchange user list
 ```
-3. Choose a id and token for your edge node, create it, and verify it:
-```
-export HZN_EXCHANGE_NODE_AUTH="PUT-ANY-NODE-ID-HERE:PUT-ANY-NODE-TOKEN-HERE"
+
+5. Choose an id and token for your edge node, create it, and verify it:
+
+```bash
+export HZN_EXCHANGE_NODE_AUTH="<choose-any-node-id>:<choose-any-node-token>"
 hzn exchange node create -n $HZN_EXCHANGE_NODE_AUTH
 hzn exchange node confirm
 ```
-4. Register your edge node with Horizon to use the helloworld pattern:
-```
+
+## <a id=using-helloworld-pattern></a> Using the Hello World Example Edge Service with Deployment Pattern
+
+1. Register your edge node with Horizon to use the helloworld pattern:
+
+```bash
 hzn register -p IBM/pattern-ibm.helloworld
 ```
-5. The edge device will make an agreement with one of the Horizon agreement bots (this typically takes about 15 seconds). Repeatedly query the agreements of this device until the `agreement_finalized_time` and `agreement_execution_start_time` fields are filled in:
-```
+
+2. The edge device will make an agreement with one of the Horizon agreement bots (this typically takes about 15 seconds). Repeatedly query the agreements of this device until the `agreement_finalized_time` and `agreement_execution_start_time` fields are filled in:
+
+```bash
 hzn agreement list
 ```
 
-6. Once the agreement is made, list the docker container edge service that has been started as a result:
```

Comment:
Once s/b After. This is a global comment. Once should be used to indicate how many times something is done..as in one time. Use after when referring to the passage of time.

Reply:
done